test/tests/functional/pbs_ralter.py

```diff
@@ -2234,3 +2235,32 @@ def test_select_fail_revert(self):
         a = {'Resource_List.select': '8:ncpus=1', 'Resource_List.ncpus': 8,
              'Resource_List.nodect': 8}
         self.server.expect(RESV, a, id=rid)
+
+    def test_resv_resc_assigned(self):
+        """
+        Test that when an ralter -D is issued, the resources on the node
+        are still correct
+        """
+
+        offset = 60
+        dur = 60
+        select = "4:ncpus=1"
+
+        rid, start, end = \
+            self.submit_and_confirm_reservation(offset, dur, select=select)
+
+        self.server.status(RESV, 'resv_nodes', id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+
+        sleepdur = start - time.time()
+        self.logger.info('Sleeping until reservation starts')
+        self.server.expect(RESV,
+                           {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')},
+                           offset=sleepdur)
+
+        self.server.alterresv(rid, {ATTR_resv_duration: '10:00'})
```

Comment:
I'd use alter_a_reservation().  It checks to make sure the alter succeeded.

Reply:
That function checks to see if the reservation is either confirmed or running. since the reservation is deleted, it fails.
I can add a log_match if the reservation alter was confirmed

Reply:
Oops, I was looking at the wrong test. I will take another look.